### PR TITLE
timeout for functions that use udp

### PR DESCRIPTION
**Pull request by misimisi171718**
Added a time to the UDPsocket class this should solve issue #38 without any memory leaks

### CODE REVIEW

File: src/structure/UDPSocket.ts [hunk 1]
```diff
@@ -93,6 +95,15 @@ class UDPSocket {
 			this.socket.on('message', messageHandler);
 			this.socket.on('error', errorHandler);
 			this.socket.on('close', closeHandler);
+
+			if(this.timeout != 0)
```
Comment (PassTheMayo) [inline src/structure/UDPSocket.ts:99]: Spacing, comparison operators, and bracket placement should follow linting rules

**Review by PassTheMayo — CHANGES_REQUESTED** [review]
This PR looks good except for a minor linting rule violation. Once this is fixed, I will accept the PR.

**Comment by misimisi171718** [comment]
I fixed the linting issue it should be good to merge

**Comment by PassTheMayo** [comment]
I had actually already implemented a timeout for UDP methods in my working copy for the latest `3.6.0` release which uses a `TimeoutPromise` utility class that was used previously. I am gonna go ahead and close this for now.

**Pull request closed without merging**